.github/actions/create-pr/action.yml

```diff
@@ -0,0 +1,92 @@
+name: "Create PR custom action"
+description: "Create a PR and a temporary branch, close duplicates"
+
+# PROCESS
+#
+# 1. Setup git client using Powertools for AWS Lambda bot username
+# 2. Pushes staged files to a temporary branch
+# 3. Creates a PR from temporary branch against a target branch (typically trunk: develop, main, etc.)
+# 4. Searches for duplicate PRs with the same title
+# 5. If duplicates are found, link to the most recent one, close and delete their branches so we keep a single PR
+# 6. In the event of failure, we delete the now orphaned branch (if any), and propagate the failure
+
+# REQUIREMENTS
+# You need to enable "Allow GitHub Actions to create and approve pull requests" in your repository settings
+# You need to have "skip-changelog" label in your repository to skip changelog generation.
+
+# USAGE
+#
+# - name: Create PR
+#   id: create-pr
+#   uses: ./.github/actions/create-pr
+#   with:
+#     files: "CHANGELOG.md"
+#     temp_branch_prefix: "ci-changelog"
+#     pull_request_title: "chore(ci): changelog rebuild"
+#     github_token: ${{ secrets.GITHUB_TOKEN }}
+# - name: Step to demonstrate how to access outputs (no need for this)
+#   run: |
+#     echo "PR number: ${PR_ID}"
+#     echo "Branch: ${BRANCH}"
+#   env:
+#     PR_ID: ${{ steps.create-pr.outputs.pull_request_id}}
+#     BRANCH: ${{ steps.create-pr.outputs.temp_branch}}
+
+inputs:
+  files:
+    description: "Files to add separated by space"
+    required: true
+  temp_branch_prefix:
+    description: "Prefix for temporary git branch to be created, e.g, ci-docs"
+    required: true
+  pull_request_title:
+    description: "Pull Request title to use"
+    required: true
+  github_token:
+    description: "GitHub token for GitHub CLI"
+    required: true
+  target_branch:
+    description: "Branch to target when creating a PR against (main, by default)"
+    required: false
+    default: main
+
+outputs:
+  pull_request_id:
+    description: "Pull request ID created"
+    value: ${{ steps.create-pr.outputs.pull_request_id }}
+  temp_branch:
+    description: "Temporary branch created with staged changed"
+    value: ${{ steps.create-pr.outputs.temp_branch }}
+
+runs:
+  using: "composite"
+  steps:
+    - id: adjust-path
+      run: echo "${{ github.action_path }}" >> $GITHUB_PATH
+      shell: bash
+    - id: setup-git
+      name: Git client setup and refresh tip
+      run: |
+        git config --global user.name 'github-actions[bot]'  
```

Comment:
we also have the PT bot user: aws-powertools-bot

Reply:
ah this is much better, will change

Reply:
Using that (non existent) user makes the UI broken
<img width="786" alt="image" src="https://github.com/aws-powertools/powertools-lambda-typescript/assets/7353869/b87c9ec1-2d22-472a-a8f5-dcb09322e28f">

Unless we create an actual bot user (aka a GH App), the user won't exist so better to use the action user which exists.

Reply:
This is a GitHub user, not a GitHub App, thus we can't use it. I will keep github-actions bot for the config.